### PR TITLE
PXB-2740 PXB displays the encryption key when the backup is decrypted…

### PR DESCRIPTION
**Pull request by rahulmalik87**
… and decompressed

https://jira.percona.com/browse/PXB-2740

Problem:
PXB displays the encryption key when the backup is decrypted and decompressed

Analysis:
PXB uses system function to call xbcrypt with encryption password to decrypt
the password. system function creates sub process. password is only
removed by sub process . The main process shows the encrypted password
in ps

Fix:
set the enviorment variable XBCRYPT_ENCRYPTION_KEY in pxb and use
that variable inside xbcrypt

### CODE REVIEW

File: storage/innobase/xtrabackup/src/backup_copy.cc [hunk 1]
```diff
@@ -2304,6 +2302,11 @@ decrypt_decompress()
 
 	ut_a(xtrabackup_parallel >= 0);
 
+	if (xtrabackup_encrypt_key) {
+		setenv("XBCRYPT_ENCRYPTION_KEY",
+				xtrabackup_encrypt_key, 1);
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/backup_copy.cc]: can be adjusted to one line?

**Review by satya-bodapati — CHANGES_REQUESTED** [review]
Looks good. 
1. Please paste the test result (if jenkins works, preferably jenkins, else manual result of testsuite).
2. Please update the 8.0 PR as well
3. Fix commit message and github description "XBCRYPT_ENCRYPTION_PASSWORD", right env variable is XBCRYPT_ENCRYPTION_KEY

File: storage/innobase/xtrabackup/src/backup_copy.cc (hunk 1 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/src/backup_copy.cc]: done

**Comment by satya-bodapati** [comment]
@rahulmalik87 please squash and fix commit message & description

Check other comments too

**Comment by rahulmalik87** [comment]
@satya-bodapati fixed the commit message.

./run.sh -t t/check_for_passwords.sh k
Using /home/rahul.malik/MySQL/build/x7/xtrabackup-test/var as test root
Detecting server version...
Running against MySQL 5.7.37-debug-log (InnoDB 5.7.37)
Using 'xtrabackup' as xtrabackup binary
Autodetected number of cores: 24
Limiting to 16 to avoid excessive resource consumption
Using 16 parallel workers

==============================================================================
TEST                                   WORKER    RESULT     TIME(s) or COMMENT
------------------------------------------------------------------------------
check_for_passwords                      w1	[passed]    37
==============================================================================
Spent 37 of 43 seconds executing testcases

SUMMARY: 1 run, 1 successful, 0 skipped, 0 failed

See results/ for detailed output
{qa-sm-twina02 x7 o7 xtrabackup-test}$

**Review by satya-bodapati — APPROVED** [review]

**Pull request merged**